src/main/kotlin/io/easybreezy/hr/api/Router.kt

```diff
@@ -318,6 +340,71 @@ class Router @Inject constructor(
         }
     }
 
+    private fun Route.eventsRouting() {
+        route("/events") {
+            authorize(setOf(Activity.EVENTS_MANAGE)) {
+                post<Response.Either<Response.Ok, Response.Errors>, Open>("") { command ->
+                    controller<EventController>(this).openEvent(resolvePrincipal<UserPrincipal>(), command)
+                }
+            }
+
+            route("/{id}") {
+                authorize(setOf(Activity.EVENTS_MANAGE), {
+                    val principal = principal<UserPrincipal>()
+                    val event = locations.resolve<ID>(this).id
+                    queryExecutor.execute(IsEventOwner(event, principal!!.id))
+                }) {
+                    post<Response.Either<Response.Ok, Response.Errors>, UpdateDetails, ID>("") { command, params ->
+                        controller<EventController>(this).updateEventDetails(params.id, command)
+                    }
+                    post<Response.Either<Response.Ok, Response.Errors>, AddParticipants, ID>("/participants") {
+                            command, params ->
+                        controller<EventController>(this).addParticipants(params.id, command)
+                    }
+                    postParams<Response.Ok, ID>("/cancel") { params ->
+                        controller<EventController>(this).cancelEvent(params.id)
+                    }
+                    post<Response.Either<Response.Ok, Response.Errors>, ChangeConditions, ID>("/conditions") {
+                            command, params ->
+                        controller<EventController>(this).updateEventConditions(params.id, command)
+                    }
+                }
+                authorize(setOf(Activity.EVENTS_LIST), {
+                    val id = principal<UserPrincipal>()!!.id
+                    val event = locations.resolve<ID>(this).id
+                    queryExecutor.execute(CanViewEvent(event, id))
+                }) {
+                    get<Response.Data<Event>, ID>("") { params ->
+                        controller<EventController>(this).event(params.id)
+                    }
+                }
+                route("/participants") {
+                    authorize(setOf(Activity.EVENTS_MANAGE), {
```

Comment:
Could I see events where I'm a participant?

Reply:
User can see open events and all events where hi is participant or owner.